### PR TITLE
Convert validation loop config warnings to `PossibleUserWarning`

### PR DESCRIPTION
**Pull request by CompRhys**
## What does this PR do?

Extends #10644 by making more warnings `PossibleUserWarning`, related to #13358 for cleaning-up/reducing stdout verbosity.

### Does your PR introduce any breaking changes? If yes, please list them.

If users are already filtering `UserWarning` code will no longer filter messages converted to `PossibleUserWarning`.

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/Lightning-AI/lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by CompRhys** [comment]
Have updated `CHANGELOG.md` but feel this is minor

**Review by rohitgr7 — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

File: src/pytorch_lightning/trainer/configuration_validator.py [hunk 1]
```diff
@@ -115,9 +116,15 @@ def __verify_train_val_loop_configuration(trainer: "pl.Trainer", model: "pl.Ligh
     has_val_step = is_overridden("validation_step", model)
 
     if has_val_loader and not has_val_step:
-        rank_zero_warn("You passed in a `val_dataloader` but have no `validation_step`. Skipping val loop.")
+        rank_zero_warn(
+            "You passed in a `val_dataloader` but have no `validation_step`. Skipping val loop.",
```
Comment (carmocca) [inline src/pytorch_lightning/trainer/configuration_validator.py:120]: The warning below can be a false positive, but when would this one be?

Comment (CompRhys) [reply src/pytorch_lightning/trainer/configuration_validator.py]: good point, it was the other warning that I was trying to suppress

File: tests/tests_pytorch/trainer/test_config_validator.py [hunk 2]
```diff
@@ -53,13 +54,13 @@ def test_fit_val_loop_config(tmpdir):
     trainer = Trainer(default_root_dir=tmpdir, max_epochs=1)
 
     # no val data has val loop
-    with pytest.warns(UserWarning, match=r"You passed in a `val_dataloader` but have no `validation_step`"):
+    with pytest.warns(PossibleUserWarning, match=r"You passed in a `val_dataloader` but have no `validation_step`"):
```
Comment (carmocca) [inline tests/tests_pytorch/trainer/test_config_validator.py:57]: You will need to revert this too

**Review by carmocca — APPROVED** [review]

**Review by otaj — APPROVED** [review]
